packages/analytics/analytics-chart/src/components/TopNTable.vue

```diff
@@ -0,0 +1,268 @@
+                  dimension: displayKey,
+                }"
+              >
+                {{ getName(entry) }}
+              </slot>
+            </div>
+            <div class="column-2">
+              &nbsp; {{ getValue(entry) }}
+            </div>
+          </div>
+        </div>
+      </div>
+    </template>
+  </KCard>
+</template>
+
+<script setup lang="ts">
+import { computed, PropType } from 'vue'
+import { AnalyticsExploreV3Result, AnalyticsExploreV3Meta, AnalyticsExploreRecord, RecordEvent } from '@kong-ui-public/analytics-utilities'
+// @ts-ignore - approximate-number no exported module
+import approxNum from 'approximate-number'
+import composables from '../composables'
+
+const props = defineProps({
+  title: {
+    type: String,
+    required: true,
+  },
+  description: {
+    type: String,
+    default: '',
+  },
+  data: {
+    type: Object as PropType<AnalyticsExploreV3Result>,
+    required: true,
+  },
+  emptyStateTitle: {
+    type: String,
+    default: '',
+  },
+})
+
+const { i18n } = composables.useI18n()
+const meta = computed((): AnalyticsExploreV3Meta => props.data.meta)
+const records = computed((): AnalyticsExploreRecord[] => props.data.records)
+const hasData = computed((): boolean => !!(records.value?.length))
+const displayKey = computed((): string => {
+  if (!meta.value?.display) {
+    return ''
+  }
+
+  return Object.keys(meta.value.display)?.[0]
+})
+const displayRecord = computed(() => {
+  if (!displayKey.value) {
+    return {}
+  }
+
+  return meta.value.display[displayKey.value]
+})
+const columnKey = computed((): string => {
+  if (!meta.value?.metricNames?.length) {
+    return ''
+  }
+
+  return meta.value.metricNames[0]
+})
+const columnName = computed((): string => {
+  if (columnKey.value) {
+    return i18n.t(`chartLabels.${columnKey.value}` as any) || columnKey.value
+  }
+
+  return ''
+})
+
+const errorMessage = computed((): string => {
+  if (hasData.value) {
+    if (!meta.value) {
+      return i18n.t('topNTable.errors.meta')
+    }
+
+    if (!Object.keys(displayRecord.value).length) {
+      return i18n.t('topNTable.errors.display')
+    }
+
+    if (!columnKey.value) {
+      return i18n.t('topNTable.errors.metricNames')
+    }
+  }
+
+  return ''
+})
+
+const getEvent = (record: AnalyticsExploreRecord): RecordEvent => {
+  return record.event
+}
+
+const getId = (record: AnalyticsExploreRecord): string => {
+  const event = getEvent(record)
+
+  return String(event[displayKey.value])
+}
+const getName = (record: AnalyticsExploreRecord): string => {
+  const id = getId(record)
+
+  return String(displayRecord.value[id]) || '–'
+}
+const getValue = (record: AnalyticsExploreRecord): string => {
+  if (!columnKey.value) {
+    return '–'
+  }
+
+  const event = getEvent(record)
+  const val = event[columnKey.value]
+
+  if (!val) {
+    return '–'
+  }
+
+  return approxNum(val, { capital: true, round: true })
+}
+</script>
+
+<style lang="scss" scoped>
```

Comment:
For all of the styles below, can you utilize `@kong/design-tokens` SCSS variables for the values that exist? e.g. `font-size`, `line-height`, `space`, etc.?